streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdater.java

```diff
@@ -506,16 +506,14 @@ private void addTask(final Task task) {
 
         private void removeTask(final TaskId taskId, final CompletableFuture<RemovedTaskResult> future) {
             try {
-                if (updatingTasks.containsKey(taskId)) {
-                    removeUpdatingTask(taskId, future);
-                } else if (pausedTasks.containsKey(taskId)) {
-                    removePausedTask(taskId, future);
-                } else {
-                    if (!removeRestoredTask(taskId, future)) {
-                        if (!removeFailedTask(taskId, future)) {
-                            future.complete(null);
-                            log.warn("Task " + taskId + " could not be removed from the state updater because "
-                                + "the state updater does not own this task.");
+                if (!removeUpdatingTask(taskId, future)) {
```

Comment:
nit: Used short-circuiting to avoid the nesting

```
if (!removeUpdatingTask(taskId, future)
 	&& !removePausedTask(taskId, future)
	&& .. 
```

Reply:
Oh, thanks! I am going to change that in one of my future PRs.